### PR TITLE
DOCSP-41140: Change Streams

### PR DESCRIPTION
**Pull request by mcmorisi**
# Pull Request Info

[PR Reviewing Guidelines](https://github.com/mongodb/docs-kotlin-sync/blob/master/REVIEWING.md)

JIRA - <https://jira.mongodb.org/browse/DOCSP-41140>
Staging - <https://preview-mongodbmcmorisi.gatsbyjs.io/kotlin-sync/DOCSP-41140-change-streams/read/change-streams/>

## Self-Review Checklist

- [x] Is this free of any warnings or errors in the RST?
- [x] Did you run a spell-check?
- [x] Did you run a grammar-check?
- [ ] Are all the links working?
- [x] Are the [facets and meta keywords](https://wiki.corp.mongodb.com/display/DE/Docs+Taxonomy) accurate?

### CODE REVIEW

File: source/read/change-streams.txt
Comment (mcmorisi) [inline source/read/change-streams.txt:192]: Note for reviewer: these links are broken at the moment.

Rea has a change that has yet to be merged that changes the `{+api+}` source constant. This will work once it's merged.

File: source/read/change-streams.txt
Comment (mcmorisi) [inline source/read/change-streams.txt:278]: Note for reviewer: these links are broken at the moment.

Rea has a change that has yet to be merged that changes the `{+api+}` source constant. These will work once it's merged.

**Review by rustagir — CHANGES_REQUESTED** [review]
a few things, good work!

File: source/read/change-streams.txt
Comment (rustagir) [inline source/read/change-streams.txt:86]: S: add highlighting to highlight the `updateDescription` array

File: source/read/change-streams.txt
Comment (rustagir) [inline source/read/change-streams.txt:134]:
```suggestion
Match Specific Events Example
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

File: source/read/change-streams.txt
Comment (rustagir) [inline source/read/change-streams.txt:137]:
```suggestion
The following example uses the ``pipeline`` parameter that includes a ``$match``
to open a change stream that records only update operations:
```

File: source/read/change-streams.txt
Comment (rustagir) [inline source/read/change-streams.txt:219]:
```suggestion
pre-image in the change stream event, pass one of the following options to
the ``fullDocumentBeforeChange()`` method:
```

File: source/read/change-streams.txt
Comment (rustagir) [inline source/read/change-streams.txt:229]:
```suggestion
The **post-image** is the full version of a document *after* a change. To include the
post-image in the change stream event, pass one of the following options to the ``fullDocument()``
method:
```

File: source/read/change-streams.txt
Comment (rustagir) [inline source/read/change-streams.txt:240]:
```suggestion
The following example calls the ``watch()`` method on a collection and includes the post-image
of updated documents in the results by specifying the ``fullDocument`` parameter:
```

File: source/read/change-streams.txt
Comment (rustagir) [inline source/read/change-streams.txt:252]: S: remove extra line

File: source/read/change-streams.txt
Comment (rustagir) [inline source/read/change-streams.txt:254]:
```suggestion
.. code-block:: none
```

For outputs

**Review by rustagir — APPROVED** [review]
approving with some non blocking comments, I can take another look if needed

File: source/includes/read/change-streams.kt
Comment (jyemin) [inline source/includes/read/change-streams.kt:21]: Why include this?  It's already on by default.

**Review by jyemin — COMMENTED** [review]
I'm not quite sure how to review this.  Is all the prose brand new or is it a copy of existing documentation for either the Java sync driver or the Kotlin coroutine driver, or some combo of the two?

File: source/includes/read/change-streams.kt
Comment (mcmorisi) [reply source/includes/read/change-streams.kt:21]: To your top-level comment (questions about how to review this) we usually request technical reviews to confirm that the technical language in the document is accurate, as well as whether the code samples are up to standard.

As for the `retryWrite` line, I can remove that.

Comment (jyemin) [reply source/includes/read/change-streams.kt]: ok, let me know when you have removed it.

Comment (mcmorisi) [reply source/includes/read/change-streams.kt]: I've removed it!

File: source/read/change-streams.txt
Comment (jyemin) [inline source/read/change-streams.txt:101]: I'm not clear on why only the updateDescription is being highlighted.  It would make sense if this section was discussing the updateDescription in particular, but it is not.  Moreover, below we show an example of filtering on the operationType, and that is not highlighted.

I suggest that the highlighting be removed, and also remove this from the top:

```
   {
     "cursor": {
       "nextBatch": [
```

and this from the bottom:

```
       ],
       ...
     }
   }
```

The reason is that the cursor/nextBatch wrapper is part of the cursor wire protocol and is never actually visible to the application.

File: source/read/change-streams.txt
Comment (jyemin) [inline source/read/change-streams.txt:180]: This is not really customization.  Rather, it's one of several methods that can be used to iterator the change stream cursor.  It can be removed from the table.

File: source/read/change-streams.txt
Comment (jyemin) [inline source/read/change-streams.txt:195]: Same as above, this is not really customization.

File: source/read/change-streams.txt
Comment (jyemin) [inline source/read/change-streams.txt:199]: This link 404s for me.  Maybe same problem as pointed out below?

**Review by jyemin — CHANGES_REQUESTED** [review]

File: source/read/change-streams.txt
Comment (mcmorisi) [reply source/read/change-streams.txt:199]: Correct, all these links will break at the moment until merged.

File: source/read/change-streams.txt
Comment (mcmorisi) [reply source/read/change-streams.txt:195]: Will remove in next revision

File: source/read/change-streams.txt
Comment (mcmorisi) [reply source/read/change-streams.txt:180]: Will remove in next revision

File: source/read/change-streams.txt
Comment (mcmorisi) [reply source/read/change-streams.txt:101]: Will apply these suggestions to the next revision!

File: source/read/change-streams.txt
Comment (jyemin) [inline source/read/change-streams.txt:97]: What are these ellipses referring to, and is it necessary to have two sets of them?

Comment (mcmorisi) [reply source/read/change-streams.txt:97]: In this case, I was simply cutting out bits from the result I received when I was testing this myself and replacing them with ellipses.

I can update it to be just one occurrence!

**Review by jyemin — APPROVED** [review]
LGTM!

**Pull request merged**
